File: 2017/2/solve.js
const fs = require('fs');
fs.readFile('input', (_, d) => {
    // parse input
    let lines = d.toString().trim().split('\n').map(
        l => l.split('\t').map(n => Number(n)).sort((a,b) => a-b)
    ); /* lines=[[10, 27, ..., 294], ...] Each inner array is sorted. */
    // part 1
    console.log(lines.map(
        l => l.slice(-1)-l[0]   // [last elem] - first elem == max-min
    ).reduce((a,b) => a+b));
    // part 2
    console.log(lines.flatMap(l => l.flatMap(
        // due to sorting, w > v is guaranteed.
        (v,i) => l.slice(i+1).filter(w => w%v == 0).map(w => w/v)
    )).reduce((a,b) => a+b));
});

File: 2017/2/solve.py
from itertools import combinations as C
with open('input') as f: lines = [list(sorted(map(int,l.strip().split('\t')))) for l in f.read().strip().split('\n')]
print(sum(l[-1]-l[0] for l in lines))
print(sum((lambda t: t[1]//t[0])(next(filter(lambda t: t[1]%t[0] == 0, C(sorted(l),2)))) for l in lines))